examples/jaeger/src/test/java/io/opentelemetry/example/JaegerExampleIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+package io.opentelemetry.example;
+
+import static io.restassured.RestAssured.given;
+
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import org.awaitility.Awaitility;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+/**
+ * Testcontainers integration test to verify that the trace generated by the Jaeger example app is
+ * stored in an instance of Jaeger.
+ *
+ * <p>A Jaeger-all-in-one container is started and a Docker image is created on-the-fly which
+ * contains the executable JAR of the example app (but not started). The test starts the example app
+ * container before verifying that the trace is received by Jaeger.
+ */
+@RunWith(JUnit4.class)
+public class JaegerExampleIntegrationTest {
+
+  private static final String ARCHIVE_NAME = System.getProperty("archive.name");
+  private static final String APP_NAME = "jaeger-example-app.jar";
+
+  private static final int QUERY_PORT = 16686;
+  private static final int COLLECTOR_PORT = 14250;
+  private static final String JAEGER_VERSION = "1.17";
+  private static final String SERVICE_NAME = "example";
+  private static final String JAEGER_HOSTNAME = "jaeger";
+  private static final String JAEGER_URL = "http://localhost";
+
+  private static final Network network = Network.newNetwork();
+
+  @ClassRule
+  public static GenericContainer jaegerContainer =
+      new GenericContainer<>("jaegertracing/all-in-one:" + JAEGER_VERSION)
+          .withNetwork(network)
+          .withNetworkAliases(JAEGER_HOSTNAME)
+          .withExposedPorts(COLLECTOR_PORT, QUERY_PORT)
+          .waitingFor(new HttpWaitStrategy().forPath("/"));
+
+  private static GenericContainer jaegerExampleAppContainer =
+      new GenericContainer(buildJaegerExampleAppImage())
+          .withNetwork(network)
+          .waitingFor(Wait.forLogMessage(".*Bye.*", 1));
+
+  private static ImageFromDockerfile buildJaegerExampleAppImage() {
```

Comment:
Consider using `withCopyFileToContainer` and `withCmd` on `GenericContainer` instead.
There is no ned to build an image here and it will only add an overhead to the startup :)

Reply:
thats a good spot! thanks!